Objects/longobject.c

```diff
@@ -114,14 +116,15 @@ PyLongObject *
 _PyLong_New(Py_ssize_t size)
 {
     PyLongObject *result;
-    if (size > (Py_ssize_t)MAX_LONG_DIGITS) {
+    Py_ssize_t size_abs = Py_ABS(size);
```

Comment:
Let's revert this one too. We can re-introduce it in a separate PR, if performance measurements show that it's worth it.